Sources/xcproj/Path+Helpers.swift

```diff
@@ -0,0 +1,16 @@
+import PathKit
+
+// MARK: - Path Extension (Extras)
+
+extension Path {
+
+    /// - Returns: Base name for xcodeproj directory
+    internal func projectName() -> String? {
```

Comment:
I'm not sure about having this method. This method is thought to be used with `.pbxproj` paths, and that's not explicit in the definition.  Since it's just calling `lastComponentWithoutExtension` if it's an xcode project I'd move that line to the places where we call this method.

Reply:
I'm starting to think that we should not automatically set project name during writing. I would guess, it is not very common use case, where someone wan't so save project with new name. Considering this, maybe it should be users responsibility to set new project name, before serialising project.

What if I move
```swift
if parent().extension == "xcodeproj" {
  return parent().lastComponentWithoutExtension
} else {
  return nil
}
```
into https://github.com/xcodeswift/xcproj/pull/152/files/15bd83a1794242f4ddc37d3ce0880c9c7207b914#diff-682f894948ca1d5648a6d9bd7593c396R120
Turning it into:
```swift
func updateProjectName(path: Path) {
  guard path.parent().extension == "xcodeproj" else {
    return
  }
  let projectName = path.parent().lastComponentWithoutExtension
  let rootProject = projects.first(where: { $0.reference == rootObject })
  rootProject?.name = projectName
}
```

And add tests to check if `public init(path: Path) throws` updates project name.

Reply:
@pepibumur thoughts?

Reply:
Good point. I don't think people would be renaming the project once created, and if they do it's up to them to change the name accordingly. I think moving the updateProjectName in the init and testing it is a good idea. Go for it @allu22

Reply:
@pepibumur i've made these changes. Would live if you could give one more look :)